store/postgres/src/connection_pool.rs

```diff
@@ -503,8 +503,17 @@ impl ConnectionPool {
     /// # Panics
     ///
     /// If any errors happen during the migration, the process panics
-    pub fn setup(&self) {
-        self.get_ready().ok();
+    pub async fn setup(&self) {
+        let pool = self.clone();
+        let err = graph::spawn_blocking_allow_panic(move || {
+            pool.get_ready().ok();
+        })
+        .await
+        .unwrap_err();
```

Comment:
Not sure on what you intended but you can propagate panics by just calling `.unwrap()`.